atef/widgets/config/window.py

```diff
@@ -294,8 +300,102 @@ def _new_tab(
         # set up edit-run toggle
         tab_bar = self.tab_widget.tabBar()
         widget.toggle.stateChanged.connect(widget.switch_mode)
+        widget.tree_view.setContextMenuPolicy(Qt.CustomContextMenu)
+        widget.tree_view.customContextMenuRequested.connect(self.context_menu)
         tab_bar.setTabButton(curr_idx, QtWidgets.QTabBar.LeftSide, widget.toggle)
 
+    def copy(self):
+        """Copy the currently selected item and its parent to the clipboard"""
+        current_item = self.get_current_tree().current_item
+        parent_data = current_item.parent().orig_data
+        item_data = current_item.orig_data
+        # make deep copies now in case items are modified after copying but before pasting
+        if isinstance(parent_data, SetValueStep):
+            item_data = parent_data.success_criteria[parent_data.children().index(item_data)]
+        self.clipboard = (deepcopy(parent_data), deepcopy(item_data))
+
+    def start_pasting(self, list_attr=None, location=None):
```

Comment:
On second read, this method is a bit confusing.  The return type is `Union[Tuple[PageWidget, AnyDataclass], AnyDataclass]`?  After which you have to inspect the return from this method to decide what to do?

The purpose of this method is to gather the data and paste location?  A good docstring here might be helpful